app/controllers/users/registrations_controller.rb

```diff
@@ -37,34 +37,25 @@ def signup_set_avatar
 
   
   def show
-    @tab = params[:tab]
-    @tab = "posts" if @tab.blank?
-    
-    @user = User.find(params[:user_id])
-    
-    @navigation_tabs = [
-      {
-        "route" => user_path(@user, tab: "posts"),
-        "text" => "Posts",
-        "active" => @tab == "posts" || @tab.blank?,
-      },
-      {
-        "route" => user_path(@user, tab: "likes"),
-        "text" => "Likes",
-        "active" => @tab == "likes",
-      },
-      {
-        "route" => user_path(@user, tab: "comments"),
-        "text" => "Comments",
-        "active" => @tab == "comments",
-      },
-    ]
-    
-    posts = Post.all.includes(:shared_posts_relation).with_attached_images.order(created_at: :desc).load_async
-    shared_posts = SharedPost.all.order(created_at: :desc).load_async
-
-    @posts_and_shared = (posts + shared_posts).sort_by { |post| post.created_at }
-    @posts_and_shared = @posts_and_shared.reverse
+    @publications = Publications.new(params[:user_id], params[:page], 5)
+    @user_profile = UserProfile.new(params[:user_id], params[:tab], @publications)
+
+    if params[:page]
+
+      case params[:tab]
+      when 'posts'
+        render turbo_stream:
+          turbo_stream.append('profile_posts_list', partial: 'users/profile-posts-list')
+      when 'likes'
+        render turbo_stream:
+          turbo_stream.append('profile_liked_posts_list', partial: 'users/profile-liked-posts-list')
+      when 'comments'
+        render turbo_stream:
+          turbo_stream.append('profile_comments_list', partial: 'users/profile-comments-list')
+      end
+
```

Comment:
Please review the code's indentation, as the initial view doesn't make it clear what you intend to achieve here.